File: workshops/Unit2/22Inheritance/22FarmSystem/src/ec/edu/espe/farmsystem/model/Chicken.java
/** Copyright ESPE-DECC
 */
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 *
 * @author Kevin Chuquimarca ESPE-DCCO
 */
public class Chicken extends FarmAnimal {

    private static boolean isMolting;

    public Chicken(int id, String breed, Date bornOn, boolean isMolting) {
        super(id, breed, bornOn);
        Chicken.isMolting = isMolting;
    }
    
    public static String putEgg(int month){
        if (month>5 & isMolting==true){
            return "The Chicken does put eggs";
        }
        return "The Chicken doesn't put eggs";
    }

    @Override
    public String toString() {
        String basicInformation = super.toString();

        String chicken = "Chicken{" + getId() + ", breed " + getBreed()
                + ", bornOn " + getBornOn() + ", isMolting " + isMolting;

        //return "Chicken{" + basicInformation + ", isMolting=" + isMolting + '}';
        return chicken;
    }

    public boolean isIsMolting() {
        return isMolting;
    }

    public void setIsMolting(boolean isMolting) {
        Chicken.isMolting = isMolting;
    }
}

File: Object-OrientedProgramming/assigments/Unit2/22CodeJavaScript/JavaScript/Chicken.js
const FarmAnimal = require('./FarmAnimal');

class Chicken extends FarmAnimal {
    #eggs;

    constructor(id,breed,eggs){
        super(id, breed);
        this.#eggs = eggs;
    }
    
    eatAnimal(food, months){
     
        return food * months;
    }

    getEggs(){
        return this.#eggs;
    }

    toString(){
       return `Chicken{id=${this.getId()}, breed=${this.getBreed()}, eggs=${this.getEggs()}}`;
    }
}

module.exports = Chicken;